extensions/notebook/src/intellisense/completionItemProvider.ts

```diff
@@ -11,22 +11,30 @@ import * as vscode from 'vscode';
 import { charCountToJsCountDiff, jsIndexToCharIndex } from './text';
 import { JupyterNotebookProvider } from '../jupyter/jupyterNotebookProvider';
 import { JupyterSessionManager } from '../jupyter/jupyterSessionManager';
-import { Deferred } from '../common/promise';
 
-const timeoutMilliseconds = 4000;
+const timeoutMilliseconds = 3000;
```

Comment:
Any specific reason form 4 to 3 seconds ?

Reply:
turns out 4 seconds is an eternity so bringing it down to improve perf in case where kernel intellisense times out. Should think about 2 seconds or less, but for remote calls this can be slow.